modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/DomainRoutes.scala

```diff
@@ -74,10 +74,13 @@ final class DomainRoutes(domains: Domains[Future], domainQueries: FilterQueries[
   protected def readRoutes(implicit credentials: Option[OAuth2BearerToken]): Route =
     (extractDomainId & pathEndOrSingleSlash) { domainId =>
       (get & authorizeResource(domainId, Read)) {
-        traceName("getDomain") {
-          onSuccess(domains.fetch(domainId)) {
-            case Some(domain) => complete(domain)
-            case None         => complete(StatusCodes.NotFound)
+        parameter('rev.as[Long].?) { revOpt =>
```

Comment:
This style of handling revision does make the code shorter but maybe a little more difficult to read and modify.

Could we consider the following style instead?
```
parameter('rev.as[Long].?) {
  case Some(rev) =>
    traceName("getDomainRevision") {
      onSuccess(...) {}
    }
  case None =>
    traceName("getDomain") {
      onSuccess(...) {}
    }
}
```